src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -496,7 +496,13 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                     }
                 }
 
-
+                var arch = await MICommandFactory.GetTargetArchitecture();
+                if (arch == TargetArchitecture.Unknown)
```

Comment:
Right now we still have target architecture in the launch options, but we are ignoring it. It seems like we should either:
1. Only ask the debugger for its target architecture if it isn't in the launch options
-or-
2. Always ask the debugger for its target architecture, but complain if it the launch options has a target architecture, and it is different from the debugger